### PR TITLE
fix: response-rewrite plugin might cause Apache AIPSIX hanging

### PR DESCRIPTION
**Pull request by Hazel6869**
### Description

https://github.com/apache/apisix/issues/7722

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by spacewander** [comment]
Please put the issue id after `Fixes #` so it can be linked.

File: t/plugin/response-rewrite.t [hunk 1]
```diff
@@ -699,3 +699,26 @@ X-A: 127.0.0.1
...
+                return
+            end
+        }
+    }
+--- request
+GET /t
+--- response_body
+
+--- no_error_log
+[error]
```
Comment (tzssangglass) [inline t/plugin/response-rewrite.t:724]: This test case would have passed even without the code changes.

File: t/plugin/response-rewrite.t [hunk 2]
```diff
@@ -699,3 +699,26 @@ X-A: 127.0.0.1
...
 [error]
+
+
+
+=== TEST 25: set an nil body with setting body_base64 to false
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.response-rewrite")
+            local ok, err = plugin.check_schema({
```
[the remark this answers is not in the document]
Comment (tokers) [reply t/plugin/response-rewrite.t:710]: @Hazel6869 You should write another case which really uses the configuration.

File: t/plugin/response-rewrite.t [hunk 3]
```diff
@@ -702,24 +702,43 @@ X-B: from 127.0.0.1 to 127.0.0.1:1980
 
 
 
-=== TEST 25: set an nil body with setting body_base64 to false
+=== TEST 25: set header(rewrite body with "")
```
Comment (tokers) [inline t/plugin/response-rewrite.t]:
```suggestion
=== TEST 25: set header and rewrite body to empty
```

**Comment by tzssangglass** [comment]
hi @Hazel6869  I think you merged the branches by mistake.

**Comment by Hazel6869** [comment]
> hi @Hazel6869 I think you merged the branches by mistake.

i will fix it, thanks

**Pull request closed without merging**